### ISSUE TITLE
Mac Address (via SSH

### ISSUE THREAD

**Issue by isarrider** [Fishwaldo/sophgo-sg200x-debian#12]
Hi,

awesome project for this tiny thing...

My DuoS gets a new IP every time it boots...
Does the MAC adress change randomly?

BR,
Alex

**Comment by Fishwaldo** [comment]
Yes. I will post some instructions soon how to setup a static MAC address.

**Comment by isarrider** [comment]
Any idea why it doesnt use the burned in one?

**Comment by Fishwaldo** [comment]
There is no one default one. 

The buildroot has some tricks to generate one on first boot and save it to the filesystem from memory (but that trick isn't compatible with Debian/Network Manager

**Comment by isarrider** [comment]
Hmm - interesting - in all my past there were always the MACs burned in...

**Comment by Fishwaldo** [comment]
It's an unfortunate trend that's becoming more prevalent. 

(There maybe a serial number in the efuse of the SOC that could be used to generate a MAC address but unfortunately no documentation exists on what efuse are present... )

**Comment by PJ5168** [comment]
Yuo can set fix mac address to "end0" in /etc/network/interfaces

**Comment by powo01** [comment]
you can set your own mac address via "nmtui" as a clone mac address

**Comment by isarrider** [comment]
which is the preferred way and how do I do that exactly?

**Comment by qgp9** [comment]
Here is my script using nmcli. At least, it works for me.

https://gist.github.com/qgp9/0d77ce769c03e2d313d72fa1b6582555
```
#!/bin/sh
DEVICE=end0
MAC="26:6E:FE:5C:CE:38" # prefered mac address

UUID=$(nmcli -f UUID,DEVICE connection show | awk -v dev=$DEVICE '$2==dev{print $1}')
nmcli connection down $UUID
nmcli connection modify $UUID ethernet.cloned-mac-address "$MAC"
nmcli connection up $UUID
```
I recommend reusing a current Mac address, as a current ssh connection is most likely alive with the same mac address and dhcp return value.

**Comment by isarrider** [comment]
Thanks...
actually this MAC stuff annoys me a bit...
Good old days where it was burnt in ;)
